### DISCUSSION TITLE
How to enable the Logs in DEBUG for Strimzi Kafka.

### DISCUSSION THREAD

**Discussion by Shaik-Rahamathulla** [strimzi/strimzi-kafka-operator#9354] [General]
How to enable the Logs in DEBUG for Strimzi Kafka in deployment or config file. We are using strimzi kafka service. I required the logs in DEBUG mode, but I'm unable to update this file log4j.properties in the config map file of kafka-service, it is printing INFO mode by default.

Could you please guide me how to change the INFO to DEBUG mode in Strimzi kafka logs.

**Comment by scholzj** [comment]
Have you read the docs? If not, please do so. If yes, please share what you tried so that we can start from there.

**Comment by Shaik-Rahamathulla** [comment]
apiVersion: v1
data:
log4j2.properties: |
name=TOConfig
appender.console.type=Console
appender.console.name=STDOUT
appender.console.layout.type=PatternLayout
appender.console.layout.pattern=%d{yyyy-MM-dd HH:mm:ss,nnnnn} %-5p [%t] %c{1}:%L - %m%n
rootLogger.level=DEBUG
rootLogger.appenderRefs=stdout
rootLogger.appenderRef.console.ref=STDOUT
rootLogger.additivity=false
logger.clients.name=org.apache.kafka.clients
logger.clients.level=INFO
logger.clients.additivity=false
logger.streams.name=org.apache.kafka.streams
logger.streams.level=INFO
logger.streams.additivity=false

monitorInterval=30
kind: ConfigMap
metadata:
creationTimestamp: "2023-11-16T09:43:13Z"
labels:
app.kubernetes.io/instance: my-cluster
app.kubernetes.io/managed-by: strimzi-cluster-operator
app.kubernetes.io/name: entity-topic-operator
app.kubernetes.io/part-of: strimzi-my-cluster
strimzi.io/cluster: my-cluster
strimzi.io/component-type: entity-topic-operator
strimzi.io/kind: Kafka
strimzi.io/name: my-cluster-entity-topic-operator
name: my-cluster-entity-topic-operator-config-1
namespace: plt-dev03
ownerReferences:

apiVersion: kafka.strimzi.io/v1beta2
blockOwnerDeletion: false
controller: false
kind: Kafka
name: my-cluster
uid: b8a72067-b554-4dc0-8101-d7655dca776b
resourceVersion: "316569255"
uid: f4be3309-6c1e-4022-ab19-d75c4f86787b

In above Configmap.yaml file we are changing the INFO to DEBUG mode and it's not reflecting in the Deployment and pods. We are changed in Deployment file of Volume mount in Custom Config file. My requirement is rootLogger.level=DEBUG should reflect in logs also. When i'm checking the logs it's still showing the INFO mode.

**Reply by scholzj** [reply]
That does not look like what the docs say. You cannot edit the internal Config Maps managed by Strimzi.

**Comment by Shaik-Rahamathulla** [comment]
Ok, But My requirement is change the logs mode into INFO to DEBUG. There is any alternate way to change it in config maps or Deployment file not by managed by strimzi.

**Reply by ppatierno** [reply]
Reinforcing what Jakub said, if you are still trying to change it on the ConfigMap, it sounds you didn't read the doc. Did you?
